packages/cashier-v2/src/lib/WithdrawalCrypto/components/WithdrawalCryptoForm/components/WithdrawalCryptoPercentageSelector/WithdrawalCryptoPercentageSelector.tsx

```diff
@@ -0,0 +1,76 @@
+import React from 'react';
+import { useFormikContext } from 'formik';
+import { Text } from '@deriv-com/ui';
+import { PercentageSelector } from '../../../../../../components/PercentageSelector';
+import { useWithdrawalCryptoContext } from '../../../../provider';
+import { TWithdrawalForm } from '../../../../types';
+import { validateCryptoInput, validateFiatInput } from '../../../../utils';
+import styles from './WithdrawalCryptoPercentageSelector.module.scss';
+
+const WithdrawalCryptoPercentageSelector: React.FC = () => {
+    const { setValues, values } = useFormikContext<TWithdrawalForm>();
+    const { accountLimits, activeAccount, fractionalDigits, getConvertedFiatAmount, isClientVerified } =
+        useWithdrawalCryptoContext();
+
+    const getPercentageMessage = (value: string) => {
+        const amount = parseFloat(value);
+        if (!activeAccount?.balance || !activeAccount.display_balance) return;
+
+        if (amount <= activeAccount.balance) {
+            const percentage = Math.round((amount * 100) / activeAccount.balance);
+            return `${percentage}% of available balance (${activeAccount.display_balance})`;
+        }
+    };
+
+    const isInvalidInput =
+        !validateCryptoInput(
+            activeAccount,
+            fractionalDigits,
+            isClientVerified,
+            accountLimits?.remainder ?? 0,
+            values.cryptoAmount
+        ) && !validateFiatInput(fractionalDigits, values.fiatAmount);
+
+    return (
+        <div className={styles['percentage-selector']}>
+            <PercentageSelector
+                amount={
+                    activeAccount?.balance &&
+                    !Number.isNaN(parseFloat(values.cryptoAmount)) &&
+                    parseFloat(values.cryptoAmount) <= activeAccount.balance
+                        ? parseFloat(values.cryptoAmount)
+                        : 0
+                }
+                balance={activeAccount?.balance ?? 0}
+                onChangePercentage={percentage => {
```

Comment:
mmm
maybe `onChangePercentage` we can be also extracted to make markup part a little bit readable ?🤔